Makefile

```diff
@@ -28,20 +31,17 @@ fix-lint: ## Fix linting errors
 	@echo "==> Fixing lint errors"
 	@./bin/golangci-lint run --fix
 
-.PHONY: build
-build: ## Build the library
-	@echo "==>"
-	@echo "==> Building the code"
-	@go build ./...
-
 .PHONY: test
 test: ## Run the tests
 	@echo "==>"
 	@echo "==> Running tests"
-	@go test ./...
+	@go test -race -cover -count=1 -coverprofile ${COVERAGE} ./...
+
+.PHONY: check
+check: test fix-lint
 
 .PHONY: all
-all: fmt lint build test ## Run all targets
+all: fmt test lint ## Run all targets
```

Comment:
we should fallow a red -> green -> refactor approach, so tests first, lint latter